packages/core/src/__tests__/ceramic-anchor.test.ts

```diff
@@ -50,18 +50,27 @@ const createCeramic = async (ipfs: IPFSApi, anchorManual: boolean, topic: string
   return ceramic
 }
 
-const anchor = async (ceramic: Ceramic): Promise<void> => {
-  await ceramic.context.anchorService.anchor()
-}
-
-const syncDoc = async (doctype: Doctype): Promise<void> => {
-  await new Promise(resolve => {
-    doctype.on('change', () => {
+const registerChangeListener = function (doc: Doctype): Promise<void> {
```

Comment:
I hit some weirdness trying to get this to work.  I originally had this as an async function, but then waiting on the returned promise in anchorDoc would hang. Removing the 'return' keyword after making it async would get it to stop hanging, but then it would fail on the next line in every test because the document log would be one entry smaller than expected.  I definitely think I'm hitting up against some rough edges of the concurrency/execution model of Node.js.  Would you be open to taking a look at this together tomorrow morning @simonovic86?

Reply:
Okay nevermind, I more or less see the problem. Making registerChangeListener async makes the compiler think that saying `await registerChangeListener(doc)` returns `void`, rather than `Promise<void>` like it should be.  I tried to change the return type to `Promise<Promise<void>>` but that doesn't seem to help, the compiler seems to flatten the stack of Promises. I even found a [bug report](https://github.com/microsoft/TypeScript/issues/27711) about it.  Anyway I think the right move is just to leave the function synchronous, as I've done here.  So I think this should be good as-is, pending your review @simonovic86

Reply:
yep, this is OK 👍